rancher/dashboard
# Can pull Helm chart app images from cluster registry or custom registry

https://github.com/rancher/dashboard/issues/6910

To test:

1. Provision two downstream RKE2 clusters, one with a private registry in the cluster config and one without.
2. In the cluster that has a registry, install a helm chart with the default settings. (I used Rancher alerting drivers) Confirm that the cluster registry from the cluster config is used in the values.yaml under `values.global.cattle.systemDefaultRegistry`. 
3. In the cluster that does not have a registry, install a helm chart with the default settings. The `values.global.cattle.systemDefaultRegistry` should be an empty string because the global default registry in the settings is also an empty string.
4. In the global settings of Rancher, change `system-default-registry` to a different string. It can be any string.
5. In the cluster that does not have a registry, install or upgrade a chart and confirm that the systemDefaultRegistry now reflects the new default registry from the global settings. Note: This is where I got stuck - the new default setting is not sent in the network request.


I have not tested the following yet, but this is what I intend to test:
6. In any cluster, install or upgrade a chart. In the first page of the helm chart install workflow, select the new checkbox that says "Use Custom Container Registry for Rancher system images"
7. Enter a custom registry
8. Confirm that the new registry is in the `values.global.cattle.systemDefaultRegistry` in the values.yaml
